gmp-commands/src/main/java/edu/gemini/aspen/gmp/commands/model/impl/ActionManagerImpl.java

```diff
@@ -157,19 +157,21 @@ private void updateClients(UpdateData updateData) {
                         //store this response to combine it with the other answers we might receive for the same action
                         _handlerResponseTracker.storeResponse(action, response);
                         if (_handlerResponseTracker.isComplete(action)) {
-                            LOG.info("Updating clients with action " + action + " response " + _handlerResponseTracker.getResponse(action));
+                            LOG.info("Updating clients with action ID " + action.getId() + " response " 
+                                    + _handlerResponseTracker.getResponse(action));
                             action.sendResponseToListeners(_handlerResponseTracker.getResponse(action));
                             //remove the action from the list of tracked actions
                             _handlerResponseTracker.removeTrackedAction(action);
 
                             //now, remove the element from the list
                             _actionList.remove(action);
                         } else {
-                            LOG.info("Received update for action " + action + " response " +
-                                    response + ". Waiting for the other parts of the action to complete...");
+                           LOG.info("Received update for action ID " + action.getId() + " response "
+                                  + response + ". Waiting for " + _handlerResponseTracker.getPendingResponses(action)
+                                  + " other parts of the action to complete...");
                             //in this case, the loop is aborted, since this action is not completed yet,
                             //so we have to keep waiting.
-                            action = null;
+                            //action = null;
```

Comment:
This is the change for the second bug. I can see this could be a problem on a multi threaded case, perhaps this code should have been done inside the lock.